### PR TITLE
Replace AWSCore with AWS.jl

### LINKED ISSUE #27: does not play nicely with AWS.jl
This package uses AWSCore.jl which does not interoperate with AWS.jl, the config struct in particular:

```julia
julia> ?CloudWatchLogStream
CloudWatchLogStream(config::AWSConfig, log_group_name, log_stream_name)

Create a reference to a CloudWatch Log Stream on AWS with the log group name and
log stream name. This constructor will automatically fetch the latest [sequence
token](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html#CWL-PutLogEvents-request-sequenceToken) for the stream.

julia> stream = CloudWatchLogStream(AWS.global_aws_config(), "my-group", "my-stream")
MethodError: no method matching CloudWatchLogStream(::AWSConfig, ::String, ::String)
Closest candidates are:
  CloudWatchLogStream(::Any, ::Any, ::Any, ::Any) at /home/ubuntu/.julia/packages/CloudWatchLogs/W1Ltg/src/stream.jl:22
  CloudWatchLogStream(::Dict{Symbol, Any}, ::String, ::String, ::Ref{Union{Nothing, String}}) at /home/ubuntu/.julia/packages/CloudWatchLogs/W1Ltg/src/stream.jl:22
  CloudWatchLogStream(::Dict{Symbol, Any}, ::AbstractString, ::AbstractString) at /home/ubuntu/.julia/packages/CloudWatchLogs/W1Ltg/src/stream.jl:36
```

I think the solution to this is to "just" upgrade to AWS.jl but I'm not sure how heavy a lift that would be.

### PR DESCRIPTION
**Pull request by ararslan**
Fixes #27.

Ping @mattBrzezinski for review. :slightly_smiling_face:

### CODE REVIEW

**Comment by ararslan** [comment]
Is there no CI on this repo?

**Comment by mattBrzezinski** [comment]
Looks like #26 never got merged, I pinged Fernando around it. Should also move this into the JuliaCloud org too.

File: Project.toml
Comment (mattBrzezinski) [inline Project.toml:3]: Guess we need to bump this to `2.0.0` now.

File: src/CloudWatchLogs.jl
Comment (mattBrzezinski) [inline src/CloudWatchLogs.jl:42]: After merging can we create an issue to call functions properly? I don't mind this to unblock but it just adds in a bit of debt.

**Review by mattBrzezinski — APPROVED** [review]
LGTM, just want to get #26 in before hand so we can test that the changes work.

File: Project.toml
Comment (ararslan) [reply Project.toml:3]: I wouldn't call this a breaking change per se. To be safe it could be 2.0 if you'd prefer, but the interface itself hasn't changed, so 1.3 would be justifiable IMO.

Comment (mattBrzezinski) [reply Project.toml:3]: We at Invenia *might* be doing this somewhere, I doubt anyone else is though. But it would break in the case of some code using, `AWS.jl`, `AWSCore.jl` and `CloudWatchLogs.jl`. Where the user would need to change passing `AWSCore.AWSConfig -> AWS.AWSConfig`.

I think also tagging as `2.0.0` is a bit safer as it won't auto update a bunch of stuff and potentially break it in scenarios like the above.

Comment (ararslan) [reply Project.toml:3]: Sure, whatever you think is best.

File: src/CloudWatchLogs.jl
Comment (ararslan) [reply src/CloudWatchLogs.jl:42]: I could do that as part of this PR, I was just being lazy...

Comment (mattBrzezinski) [reply src/CloudWatchLogs.jl:42]: Hehe, honestly it's your call if you want to spend the time. I'm fine with this wrapper function here.

Comment (ararslan) [reply src/CloudWatchLogs.jl]: Replaced

File: src/stream.jl
Comment (ararslan) [inline src/stream.jl:169]:
```suggestion
        @mock CloudWatch_Logs.describe_log_streams(
```

**Comment by ararslan** [comment]
@mattBrzezinski, it might be good to have you re-review if you get the time, since the refactoring to remove the internal function that mimics the old AWSCore API is a larger change introduced since your approval.

**Review by mattBrzezinski — APPROVED** [review]
The changes LGTM! Thanks for simplified the test files too :)
Just need to get around to #26 before this will go in. It just seems to be a timing issue like almost all the AWS CI errors happen to be. Might just merge that in anyways.

**Comment by mattBrzezinski** [comment]
Oh this is awkward... I don't remember how GitHub handles the introduction of CI workflows for merge requests that exist before those going in.

I *believe* a rebase onto `master` should resolve this and kick it off? Do you mind trying that out @ararslan , otherwise we may need to close and re-open a PR off of this branch.

**Comment by ararslan** [comment]
I rebased on master but it looks like running CI requires approval by someone with commit access.

**Comment by ararslan** [comment]
_Sigh_, needs another approval. Sorry :grimacing:

**Comment by mattBrzezinski** [comment]
> _Sigh_, needs another approval. Sorry 😬

No worries, kind of silly it needs to be done after I approved it once before, oh well.

**Comment by ararslan** [comment]
Hm. Seems something still isn't quite right. :thinking:

**Comment by iamed2** [comment]
Could this be failing because this doesn't have bors, so it's running without access to our secrets since this is a PR from a fork?

**Comment by iamed2** [comment]
@ararslan I've invited you as a collaborator with Write access so you could re-push this PR on the repo itself.

**Comment by ararslan** [comment]
Quite possibly! Thanks for the invite, I'll close this and reopen from a branch.

**Comment by mattBrzezinski** [comment]
> Could this be failing because this doesn't have bors, so it's running without access to our secrets since this is a PR from a fork?

This might be an issue w/ the high-level AWS functions. I'm looking into it now.

**Pull request closed without merging**
